shared/src/commonMain/kotlin/com/tajemniktv/tajsos/calendar/IcsCalendarProvider.kt

```
@@ -53,12 +53,19 @@ class IcsCalendarProvider(
-            if (e is CancellationException) throw e
-            println("ICS fetch failed for providerId=${provider.id}: $e")
-        }.getOrDefault(emptyList())
+        }
     }
 
+    private inline fun safeFetch(providerId: Long, block: () -> List<CalendarEventEntity>): List<CalendarEventEntity> =
+        try {
+            block()
+        } catch (e: CancellationException) {
+            throw e
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)